solutions/cses-1163.mdx

```diff
@@ -105,3 +107,111 @@ int main(){
 	return 0;
 }
 ```
+
+</CPPSection>
+<JavaSection>
+## Implementation
+[Unofficial Editorial (C++)](https://codeforces.com/blog/entry/83295)
+
+**Time Complexity:** $\mathcal{O}(n)$
+
+For each traffic light that is added, we want to find the largest current gap
+between traffic lights.
+
+First, let's get our values. We'll use a set to represent the street itself, its
+starting and ending points, and every traffic light along it. We'll also use an
+integer array to store and read the positions of the traffic lights, since we
+can't read values from sets in Java. We'll store our outputs in another array
+too.
```

Comment:
ok so this might be your first editorial, very cool!
usually in editorials you just talk about the algorithm, only talking about math and sometimes abstract data strucutres
so yeah just
```suggestion
```